Source/AwsCommonRuntimeKit/auth/signing/Signer.swift

```diff
@@ -72,6 +72,82 @@ public class Signer {
             }
         }
     }
+
+    /// Signs a body chunk according to the supplied signing configuration
+    /// - Parameters:
+    ///   - chunk: Chunk to sign
+    ///   - previousSignature: The signature of the previous component of the request: either the request itself for the first chunk,
+    ///                        or the previous chunk otherwise.
+    ///    - config: The `SigningConfig` to use when signing.
+    ///   - allocator: (Optional) allocator to override
+    /// - Returns: Signature of the chunk
+    /// - Throws: CommonRunTimeError.crtError
+    public static func signChunk(chunk: Data,
+                                 previousSignature: Data,
+                                 config: SigningConfig,
+                                 allocator: Allocator = defaultAllocator) async throws -> Data {
+        let iStreamCore = IStreamCore(iStreamable: ByteBuffer(data: chunk), allocator: allocator)
+        guard let signable = previousSignature.withAWSByteCursorPointer({ previousSignatureCursor in
+            aws_signable_new_chunk(allocator.rawValue, iStreamCore.rawValue, previousSignatureCursor.pointee)
+        }) else {
+            throw CommonRunTimeError.crtError(.makeFromLastError())
+        }
+        defer {
+            aws_signable_destroy(signable)
+        }
+
+        return try await sign(config: config, signable: signable, allocator: allocator).data(using: .utf8)!
+    }
+
+    /// Signs trailing headers according to the supplied signing configuration
+    /// - Parameters:
+    ///   - headers: list of headers to be sent in the trailer.
+    ///   - previousSignature: The signature of the previous component of the request: either the request itself for the first chunk,
+    ///                        or the previous chunk otherwise.
+    ///   - config: The `SigningConfig` to use when signing.
+    ///   - allocator: (Optional) allocator to override
+    /// - Returns: Signing Result
+    /// - Throws: CommonRunTimeError.crtError
+    public static func signTrailerHeaders(headers: [HTTPHeader],
+                                          previousSignature: Data,
+                                          config: SigningConfig,
+                                          allocator: Allocator = defaultAllocator) async throws -> Data {
```

Comment:
```suggestion
                                          previousSignature: String,
                                          config: SigningConfig,
                                          allocator: Allocator = defaultAllocator) async throws -> String {
```

Reply:
I expected it to be `Data` from API point of view.

@graebm are you suggesting this for consistency among other language bindings?

To some degree I don't like converting String to Data if the underlying implementation doesn't work on bytes array. We are wasting CPU cycles here (not significant but still).

Thoughts?

Reply:
Didn't see https://github.com/awslabs/aws-crt-swift/pull/151#discussion_r1093877914

Happy to go with String.

Reply:
Thanks, I have updated it to String.